cpu/stm32f4/include/periph_cpu.h

```diff
@@ -112,7 +112,7 @@ void gpio_init_af(gpio_t pin, gpio_af_t af);
  */
 static inline void dma_poweron(int stream)
 {
-    if (stream < 8) {
+    if (stream <= 8) {
```

Comment:
hm, as you said, the streams run from 0-7 per peripheral, but putting `<= 8` here would include them from [0-8]...